resources/js/components/Room/TokensComponent.vue

```diff
@@ -0,0 +1,397 @@
+        <b-form-radio
+          v-model.number="model.role"
+          value="1"
+        >
+          <b-badge class="text-white" variant="success">
+            {{ $t('rooms.tokens.roles.participant') }}
+          </b-badge>
+        </b-form-radio>
+        <b-form-radio
+          v-model.number="model.role"
+          value="2"
+        >
+          <b-badge variant="danger">
+            {{ $t('rooms.tokens.roles.moderator') }}
+          </b-badge>
+        </b-form-radio>
+        <template slot='invalid-feedback'><div v-html="fieldError('role')"></div></template>
+      </b-form-group>
+      <small>
+        {{ $t('rooms.tokens.editDescription') }}
+      </small>
+    </b-modal>
+  </div>
+</template>
+
+<script>
+import Base from '../../api/base';
+import { mapGetters, mapState } from 'vuex';
+import Can from '../Permissions/Can';
+import PermissionService from '../../services/PermissionService';
+import FieldErrors from '../../mixins/FieldErrors';
+import env from '../../env';
+import _ from 'lodash';
+
+export default {
+  mixins: [FieldErrors],
+  components: { Can },
+
+  props: {
+    room: Object,
+
+    modalStatic: {
+      type: Boolean,
+      default: false,
+      required: false
+    }
+  },
+
+  data () {
+    return {
+      actionRunning: false,
+      currentPage: 1,
+      errors: {},
+      isBusy: false,
+      tokens: [],
+      model: {
+        token: null,
+        firstname: null,
+        lastname: null,
+        role: null
+      }
+    };
+  },
+
+  methods: {
+    /**
+     * Copies the room link for the personalized token to the users clipboard.
+     */
+    copyPersonalizedRoomLink (token) {
+      this.$clipboard(process.env.MIX_FRONTEND_BASE_URL + this.$router.resolve({ name: 'rooms.view', params: { id: this.room.id, token: token.token }}).route.fullPath);
+      this.flashMessage.info({ message: this.$t('rooms.tokens.roomLinkCopied', { firstname: token.firstname,lastname: token.lastname }) });
+    },
+
+    /**
+     * Sends a request to delete a new token.
+     */
+    deletePersonalizedToken (bvModalEvt) {
+      bvModalEvt.preventDefault();
+
+      this.actionRunning = true;
+
+      Base.call(`rooms/${this.room.id}/tokens/${this.model.token}`, {
+        method: 'delete'
+      }).catch((error) => {
+        Base.error(error, this.$root);
+      }).finally(() => {
+        this.resetModel();
+        this.$refs['delete-token-modal'].hide();
+        this.actionRunning = false;
+        this.reload();
+      });
+    },
+
+    /**
+     * Sends a request to the server to create a new token or edit a existing.
+     */
+    savePersonalizedToken (bvModalEvt) {
+      bvModalEvt.preventDefault();
+
+      this.actionRunning = true;
+
+      const config = {
+        method: this.model.token == null ? 'post' : 'put',
+        data: _.cloneDeep(this.model)
+      };
+
+      Base.call(`rooms/${this.room.id}/tokens/${this.model.token == null ? '' : this.model.token}`, config).then(response => {
```

Comment:
if this.model.token == null the url is `rooms/ROOM-ID/tokens/ `and not `rooms/ROOM-ID/tokens` The additional tailing slash results in a redirect and fails as a result

That might work
` Base.call(`rooms/${this.room.id}/tokens${this.model.token == null ? '' : '/'+this.model.token}`, config)`

Reply:
I cannot reproduce that!

Reply:
ok, I'll have another look once the PR is done. It may just have been an issue with my local setup.